sdk/include/opentelemetry/sdk/common/attribute_utils.h

```diff
@@ -21,6 +38,12 @@ using OwnedAttributeValue = nostd::variant<bool,
                                            uint64_t,
                                            double,
                                            std::string,
+#ifdef HAVE_CSTRING_TYPE
+                                           const char *,
```

Comment:
Why do we need this? I never see us storing a `const char*` in this variant (as far as I can see, we always convert to `std::string`).

Reply:
Yeah at the moment, I don't think it's explicitly being used by `log_record` or `span_data`. Maybe there would be a future use? Not sure what the macro `HAVE_CSTRING_TYPE` will be used for. This was added in PR #374. (--cc @maxgolov)

Reply:
This would also need to be changed in [attribute_value.h](https://github.com/open-telemetry/opentelemetry-cpp/blob/master/api/include/opentelemetry/common/attribute_value.h#L21)

Reply:
Yes, it exists in `AttributeValue`. But `AttributeConverter` converts it to a `std::string` when an `AttributeValue` is converted to an `OwnedAttributeValue`, so the `const char*` slot in `OwnedAttributeValue` will never get used.

Reply:
Gotcha, removed this from `OwnedAttributeValue`